log/logger.go

```diff
@@ -1,12 +1,89 @@
 package log
 
-import cmtlog "github.com/cometbft/cometbft/libs/log"
+import (
+	"os"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+// Defines commons keys for logging
+const ModuleKey = "module"
+
+// ContextKey is used to store the logger in the context
+var ContextKey struct{}
 
-// Logger is the interface that wraps the basic logging methods.
 type Logger interface {
-	Debug(msg string, keyvals ...interface{})
-	Info(msg string, keyvals ...interface{})
-	Error(msg string, keyvals ...interface{})
+	Info(msg string, keyVals ...interface{})
+	Error(msg string, keyVals ...interface{})
+	Debug(msg string, keyVals ...interface{})
+	With(keyVals ...interface{}) Logger
+
+	Level(lvl zerolog.Level) Logger
+	Impl() *zerolog.Logger
+}
+
+type ZeroLogWrapper struct {
+	*zerolog.Logger
+}
+
+// NewNopLogger returns a new logger that does nothing
+func NewNopLogger() Logger {
+	logger := zerolog.Nop()
+	return ZeroLogWrapper{&logger}
+}
+
+// NewLogger returns a new logger
+func NewLogger() Logger {
+	output := zerolog.ConsoleWriter{Out: os.Stdout, TimeFormat: time.Kitchen}
+	logger := zerolog.New(output).With().Timestamp().Logger()
+	return ZeroLogWrapper{&logger}
+}
+
+// NewLoggerWithKV returns a new logger with the given key/value pair
+func NewLoggerWithKV(key, value string) Logger {
```

Comment:
whats the different between this and NewLogger().With()?

Reply:
No difference. It is simply a one-liner to instantiate the logger, instead of having to call with right after.